FIXES. Plenty of website fixes - Happy New Year

MikeZeDev opened the pull request:
Fixes https://github.com/manga-download/hakuneko/issues/4579, Fixes https://github.com/manga-download/hakuneko/issues/4978, Fixes https://github.com/manga-download/hakuneko/issues/5048, Fixes https://github.com/manga-download/hakuneko/issues/4268, Fixes https://github.com/manga-download/hakuneko/issues/4652, Fixes https://github.com/manga-download/hakuneko/issues/5181, Fixes https://github.com/manga-download/hakuneko/issues/5086, Fixes https://github.com/manga-download/hakuneko/issues/5153, Fixes https://github.com/manga-download/hakuneko/issues/5154, Fixes https://github.com/manga-download/hakuneko/issues/5151, Fixes https://github.com/manga-download/hakuneko/issues/4983 Fixes https://github.com/manga-download/hakuneko/issues/4532, Fixes https://github.com/manga-download/hakuneko/issues/4172, Fixes https://github.com/manga-download/hakuneko/issues/4994 , Fixes https://github.com/manga-download/hakuneko/issues/5191 Fixes https://github.com/manga-download/hakuneko/issues/5314

MikeZeDev committed:
Various Website Fixes

Fixes https://github.com/manga-download/hakuneko/issues/4579, Fixes https://github.com/manga-download/hakuneko/issues/4978, Fixes https://github.com/manga-download/hakuneko/issues/5048, Fixes https://github.com/manga-download/hakuneko/issues/4268, Fixes https://github.com/manga-download/hakuneko/issues/4652, Fixes https://github.com/manga-download/hakuneko/issues/5181, Fixes https://github.com/manga-download/hakuneko/issues/5086, Fixes https://github.com/manga-download/hakuneko/issues/5153, Fixes https://github.com/manga-download/hakuneko/issues/5154, Fixes https://github.com/manga-download/hakuneko/issues/5151, Fixes https://github.com/manga-download/hakuneko/issues/4983

MikeZeDev committed:
Another bunch of websites fixes

Fixes https://github.com/manga-download/hakuneko/issues/4532, Fixes https://github.com/manga-download/hakuneko/issues/4172, Fixes https://github.com/manga-download/hakuneko/issues/4994 , Fixes https://github.com/manga-download/hakuneko/issues/5191

MikeZeDev committed:
lint

MikeZeDev committed:
Update kuaikanmanhua.mjs

Sheepux commented on src/web/mjs/connectors/KomikTap.mjs:
```diff
@@ -7,11 +7,11 @@ export default class KomikTap extends WordPressMangastream {
         super.label = 'KomikTap';
         this.tags = [ 'manga', 'webtoon', 'hentai', 'indonesian' ];
-        this.url = 'https://komiktap.in';
+        this.url = 'https://92.87.6.124';
```
IP shouldn't be hardcoded, just initialize `this.url` in the contructor by checking the http redirect
Same for canHandleURI, should be able to dynamically include the url pattern with the IP found
The request class has some redirect features, not sure if it can help

Sheepux commented on src/web/mjs/connectors/PlotTwistNoFansub.mjs:
```diff
@@ -11,8 +9,47 @@ export default class PlotTwistNoFansub extends WordPressClarityMangaReader {
\ No newline at end of file
+    async _getChapterList( manga, callback ) {
+        let uri = new URL( manga.id, this.url );
+        let request = new Request(uri.href, this.requestOptions );
```
strange mix of async & callback 
it should be async _getChapter (instead of list) and then just return

MikeZeDev replied:
Its because the template is still old school code. Not sure something wont go wrong if i use new style.
Well, gotta rewrite it

Sheepux replied:
new style is just removing callback, add async in front, remove List from the function name  and using a return :)

Sheepux commented on src/web/mjs/connectors/ReaperScans.mjs:
```diff
@@ -11,8 +10,142 @@ export default class ReaperScans extends WordPressMadara {
+        }
+        ];
+        return {
+            payload : wiredata, token :token, referrer : referrer};
```
lint, missing new indented line before `}`

MikeZeDev replied:
Lint processor (from the PR) says there is no pb.  Copied eslint rules from the repo and checked by myslef, and no pb on my end too.
I dont get it

Sheepux replied:
the closing bracket is not at the same level as the return, that's a code formating issue (lint), but it looks like it doesn't trigge rthe lint validator

Sheepux replied:
```
        return {
            payload : wiredata, token :token, referrer : referrer};
```
to
```
        return {
            payload : wiredata, token :token, referrer : referrer
        };
```

Sheepux commented on src/web/mjs/connectors/ReaperScans.mjs:
```diff
@@ -11,8 +10,142 @@ export default class ReaperScans extends WordPressMadara {
+    }
+    getChaptersFromDoc(doc) {
+        let chapterList = [];
+        //get chapters
```
comment can be removed, function is self explained

Sheepux commented on src/web/mjs/connectors/ReaperScansBR.mjs:
```diff
@@ -1,15 +1,89 @@
+        let script = `
+        new Promise(resolve => {
+            let pages = __NEXT_DATA__;
+            resolve(pages);
```
variable should be `const`, also the variable itself is not usefull here :)

Sheepux commented on src/web/mjs/connectors/kuaikanmanhua.mjs:
```diff
@@ -20,82 +20,70 @@ export default class Kuaikanmanhua extends Connector {
+        if (chapterList.length < 1) {
+            chapterList = await this.resolveChapterList(manga, 'comicList');
         }
+        if (chapterList.length < 1) {
```
Duplicate if sequence, i'd push the  if within the other if (easier to read)
Also use an inline
`if (chapterList.length < 1) console.error('No chapters found !', this);`

Sheepux commented on src/web/mjs/connectors/kuaikanmanhua.mjs:
```diff
@@ -20,82 +20,70 @@ export default class Kuaikanmanhua extends Connector {
+    async _getChapters(manga) {
+        let chapterList = await this.resolveChapterList(manga, 'comics');
+        if (chapterList.length < 1) {
+            chapterList = await this.resolveChapterList(manga, 'comicList');
```
I feel that those double calls should be done within a single call to resolveChapterList (in the fetchUI script . And then you can just write the call here)

Sheepux commented on src/web/mjs/connectors/kuaikanmanhua.mjs:
```diff
@@ -20,82 +20,70 @@ export default class Kuaikanmanhua extends Connector {
+        let request = new Request(this.url + manga.id, this.requestOptions);
+        chapterList = await Engine.Request.fetchUI(request, script);
+
+        return chapterList;
```
unneeded variable `chapterList `? (the initial [] assignment is overwritten)
`return await Engine.Request.fetchUI(request, script);`

MikeZeDev committed:
Plottwist reformatting code

MikeZeDev committed:
Update PlotTwistNoFansub.mjs

MikeZeDev committed:
Update ReaperScansBR.mjs

MikeZeDev committed:
Update kuaikanmanhua.mjs

MikeZeDev committed:
Update ReaperScans.mjs

MikeZeDev committed:
Update KomikTap.mjs

Sheepux commented on src/web/mjs/connectors/KomikTap.mjs:
```diff
@@ -7,11 +7,19 @@ export default class KomikTap extends WordPressMangastream {
+    }
     canHandleURI(uri) {
-        return /komiktap\.in|194\.233\.66\.232|92\.87\.6\.124/.test(uri.hostname);
+        return new RegExp(this.hostPattern).test(uri.hostname);
```
host needs to be escaped before processing as a regexp as the dot is meaningfull :)
but ...
just use string.includes() without regexp.
Edit: ah the regexp was usefull for double domain matching ... using a double string.includes with an OR statement is probably the easier solution.

MikeZeDev replied:
Using a string pattern, dots in ip or in host doesnt need to be escaped. 
Direct regex pattern (with /mypattern/) needs escaping.

I tested it before posting, but ill switch to includes. 
FYI, dual regexp 'komiktap\.in|194\.233\.66\.232' was recognised by clipboard but nothing worked with komiktap\.in urls in entered. Better just using only one host.

MikeZeDev replied:
Used includes, and had to used async cause stuff is not always initializated before. Its better than calling initializate() and it works.

Sheepux replied:
you cannot switch to async "just like that" :)
the connector code is using it as a synchronous call
https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/system/ClipboardConnector.mjs#L51

MikeZeDev replied:
You think it would be better to call initialize? fine then

Sheepux replied:
lint error ;)

MikeZeDev replied:
https://smallpetselect.com/wp-content/uploads/2016/06/bunniez.hubpages.com_.png

MikeZeDev committed:
Changes on Komiktap

Have to put async before canHandleURI, otherwise _initializeConnector wasnt called before and because of that canHandleURI failed.

MikeZeDev committed:
Add ReapearScansID

https://github.com/manga-download/hakuneko/issues/5314

MikeZeDev committed:
Update KomikTap.mjs

MikeZeDev committed:
Update KomikTap.mjs

MikeZeDev committed:
Update KomikTap.mjs

MikeZeDev committed:
Update KomikTap.mjs
